repo.yml

```diff
@@ -1 +1,4 @@
-type: balena
\ No newline at end of file
+type: balena
+upstream:
+  - repo: 'balena-netdata'
+    url: 'https://github.com/balena-io-playground/balena-netdata'
```

Comment:
I take the repo is private so far? I tried looking it up and not found error popped up :open_mouth:

Reply:
Yep, it's a work-in-progress that we're hoping to get ready shortly!

Reply:
@ptrm I've open-sourced that repo, if you'd like to give this netdata approach a spin I'd love to hear what you think!

Reply:
@ptrm i wanted to follow up here and see if you had a chance to test this PR locally?

Reply:
Apologies, busy time. I'm deploying it now on my 4gb rpi4. I will push it to others if it succeeds, though I assune longer-term run would be needed to evaluate it better, right?

Reply:
It looks good in itself and fast enough on my 4gb rpi4 :)

When it comes to other devices, 1gb rpi3 runs fine. I get reboot loops on my two 2gb rpi4s. I have to investigate though, because they were overclocked to 2,1GHz to speed up the one task limit allowed on them, so not a clean environment ;) When booted with monitor plugged, no warning overlays showed though, so might be something else.

![image](https://user-images.githubusercontent.com/366258/83742691-0470f300-a65a-11ea-9339-cf7ca1550020.png)
The memory and cpu footprint worry me a bit though when it comes to 1GB devices, I think it might be good to disable netdata there, and on 2GB devices, if there ever appears a way to run 2+ tasks, it might be worth considering too.

But having said that, I don't have much idea whether it is possible to e.g. enable headless mode (so a lan peer could gather data), or change the kind and amount of metrics gathered to lower netdata's footprint.

Reply:
Also, I don't know how much this PR was considered as an aid in investigatring #47, but does netdata support capturing `oom_kill` events? One of my suspicions is that the kernel might target wrong process (e.g. sone balenaOS vital one) when out of memory, and as a consequence we get a reboot. I will try to fiddle with it in other ways, too, this month.

Reply:
@ptrm these are great notes, thank you very much for taking the time to test so thoroughly! I have opened a ticket to disable some features if we're in a low-mem/CPU situation: https://github.com/balena-io-examples/balena-netdata/issues/10.

With respect to OOM events, we _should_ collect all data up until the kernel pauses netdata as part of the OOM traversal. Unfortunately we'll lose any data for the time that it takes the kernel to traverse the page table. It would be worth reproducing https://github.com/balenalabs/rosetta-at-home/issues/47 with netdata enabled to get a better idea of what's going on!